test/cppsuite/test_harness/workload_validation.h

```diff
@@ -0,0 +1,409 @@
+                collections[it_collections.first] = nullptr;
+            }
+        }
+    }
+
+    /*
+     * collection_name is the collection that contains the operations on the different collections
+     * during the test.
+     */
+    const std::vector<std::string>
+    parse_schema_tracking_table(WT_SESSION *session, const std::string &collection_name)
+    {
+        WT_CURSOR *cursor;
+        const char *key_collection_name;
+        int key_timestamp, value_operation_type;
+        std::vector<std::string> created_collections;
+
+        testutil_check(session->open_cursor(session, collection_name.c_str(), NULL, NULL, &cursor));
+
+        while (cursor->next(cursor) == 0) {
+            testutil_check(cursor->get_key(cursor, &key_collection_name, &key_timestamp));
+            testutil_check(cursor->get_value(cursor, &value_operation_type));
+
+            debug_info(
+              "Collection name is " + std::string(key_collection_name), _trace_level, DEBUG_INFO);
+            debug_info("Timestamp is " + std::to_string(key_timestamp), _trace_level, DEBUG_INFO);
+            debug_info("Operation type is " + std::to_string(value_operation_type), _trace_level,
+              DEBUG_INFO);
+
+            if (static_cast<tracking_operation>(value_operation_type) ==
+              tracking_operation::CREATE) {
+                created_collections.push_back(key_collection_name);
+            } else if (static_cast<tracking_operation>(value_operation_type) ==
+              tracking_operation::DELETE_COLLECTION) {
+                created_collections.erase(std::remove(created_collections.begin(),
+                                            created_collections.end(), key_collection_name),
+                  created_collections.end());
+            }
+        }
+
+        return created_collections;
+    }
+
+    /*
+     * Parse the tracked operations to build a representation in memory of the collections at the
+     * end of the test. tracking_collection_name is the tracking collection used to save the
+     * operations performed on the collections during the test. collection_name is the collection
+     * that needs to be represented in memory.
+     */
+    void
+    parse_operation_tracking_table(WT_SESSION *session, const std::string &tracking_collection_name,
+      const std::string &collection_name,
+      std::map<std::string, std::map<int, std::string *> *> &collections)
+    {
+        WT_CURSOR *cursor;
+        int error_code, exact, key, key_timestamp, value_operation_type;
+        const char *key_collection_name, *value;
+
+        testutil_check(
+          session->open_cursor(session, tracking_collection_name.c_str(), NULL, NULL, &cursor));
+
+        /* Our keys start at 0. */
+        cursor->set_key(cursor, collection_name.c_str(), 0);
+        error_code = cursor->search_near(cursor, &exact);
+
+        /* As we don't support deletion, exact needs to be true. */
+        testutil_check(exact == false);
```

Comment:
Instead of false here it makes more sense to be 0, as exact can either be -1, 0, or 1.